client/hooks/use-report-export.ts

```diff
@@ -0,0 +1,202 @@
+/**
+ * External dependencies
+ */
+import { useEffect, useRef, useState } from 'react';
+import apiFetch from '@wordpress/api-fetch';
+import { useDispatch } from '@wordpress/data';
+import { __, sprintf } from '@wordpress/i18n';
+import { recordEvent } from 'wcpay/tracks';
+
+interface ExportResponse {
+	export_id?: string;
+}
+
+interface ExportURLResponse {
+	status: string;
+	download_url?: string;
+}
+
+const maxRetries = 5;
+
+/**
+ * Hook for requesting and polling for a CSV export. E.g. Transactions, Payouts, Disputes.
+ *
+ * @example
+ * const { requestReportExport, isDownloading } = useReportExport();
+ * requestReportExport( {
+ * 	exportRequestURL: '/wc/v3/payments/transactions/download?queryParam=value',
+ * 	exportFileAvailabilityEndpoint: '/wc/v3/payments/transactions/download',
+ * 	userEmail: 'test@example.com',
+ * } );
+ */
+export const useReportExport = () => {
+	const [ isDownloading, setIsDownloading ] = useState( false );
+	const { createNotice } = useDispatch( 'core/notices' );
+	const timeoutIdRef = useRef< NodeJS.Timeout | null >( null );
+	const retryCountRef = useRef( 0 );
+
+	useEffect( () => {
+		// Cancel the timeout if the component unmounts.
+		return () => {
+			if ( timeoutIdRef.current ) {
+				clearTimeout( timeoutIdRef.current );
+			}
+		};
+	}, [ timeoutIdRef ] );
+
+	interface PollForFileProps {
+		/**
+		 * The URL to check for the exported file. E.g. "/wc/v3/payments/transactions/download/{export_id}"
+		 */
+		checkFileURL: string;
+		/**
+		 * The email address the export will be sent to.
+		 */
+		userEmail: string;
+		/**
+		 * The interval in milliseconds to poll the server.
+		 */
+		interval?: number;
+	}
+
+	/**
+	 * Polls the server to check if the exported file is ready.
+	 */
+	async function pollForFile( {
+		checkFileURL,
+		userEmail,
+		interval = 1000,
+	}: PollForFileProps ) {
+		timeoutIdRef.current = setTimeout( async () => {
+			retryCountRef.current++;
+			let exportedFileURLResponse;
+			try {
+				exportedFileURLResponse = await apiFetch< ExportURLResponse >( {
+					path: checkFileURL,
+					method: 'GET',
+				} );
+			} catch ( error ) {
+				// If there's an error (like 500), continue with retry logic
+				exportedFileURLResponse = { status: 'error' };
+			}
+
+			if (
+				'success' === exportedFileURLResponse.status &&
+				exportedFileURLResponse.download_url &&
+				exportedFileURLResponse.download_url.includes(
+					'exports.wordpress.com'
```

Comment:
Why check `exports.wordpress.com` here? 

- Presumably this breaks local testing.
- If this is a security thing, can you add comment to explain why?

Reply:
> If this is a security thing

Actually that comment is useful either way!

Reply:
> Presumably this breaks local testing.
Does not break local testing, as exports.wordpress.com is hardcoded for every environment - https://github.com/Automattic/transact-platform-server/blob/1efcf99114e6539942d8e9738ea93a6e3638f9aa/server/wp-content/rest-api-plugins/endpoints/wcpay/service/class-export-service.php#L24

> security thing
Makes sure that we are downloading only a file from exports.wordpress.com, which would be placed there by the server. @Jinksi added this code I believe, do you have any suggestions?

Reply:
> @Jinksi added this code I believe, do you have any suggestions?

Git blame shows @jessy-p 😉 in https://github.com/Automattic/woocommerce-payments/commit/af1565e317e23fc8f4b82e4a1adffdeb967b3452.

I think it's fine to check only for `download_url` in the response, but agree that a comment would be helpful otherwise.

Reply:
Oh but then this
> Deleting the exports.wordpress.com seems to work, I can test locally now! I suspect there is a good reason for this check though.

I guess we can remove the check then.

Reply:
I'd like to better understand the motivation / need for the check, if it's important for real users (e.g. safety/security) then we should find a way to keep it AND allow flexible local testing 😁

Reply:
I've removed the check. I don't think its important for real users, the URL is returned from an API call that uses all our authentication mechanism like any other API from the WooCommerce Payments dashboard.